src/ray/core_worker/actor_manager.cc

```diff
@@ -91,6 +91,16 @@ std::pair<std::shared_ptr<const ActorHandle>, Status> ActorManager::GetNamedActo
       RAY_LOG(ERROR) << error_str;
       return std::make_pair(nullptr, Status::TimedOut(error_str));
     }
+  } else {
+    // When the named actor is already cached, the reference of actor_creation_return_id
+    // must be increased, so we call AddNewActorHandle again here to ensure that.
+    auto actor_handle = std::make_unique<ActorHandle>(*GetActorHandle(actor_id));
+    AddNewActorHandle(
+        std::move(actor_handle),
+        GenerateCachedActorName(actor_handle->GetNamespace(), actor_handle->GetName()),
+        call_site,
+        caller_address,
+        /*is_detached*/ true);
```

Comment:
BTW, I don't know why this `is_detached` is always `true` here, I just copy the code above. cc @rkooo567  @stephanie-wang

Reply:
I think the semantics of `is_detached` here should be "do not own the actor".